src/components/SearchForm.tsx

```diff
@@ -7,28 +7,36 @@ import CategoryPicker from './CategoryPicker';
 import Typography from '@material-ui/core/Typography';
 import Button from '@material-ui/core/Button';
 import { makeStyles } from '@material-ui/core/styles';
+import SearchResults from './SearchResults';
 
 export default function SearchForm(): JSX.Element {
   const classes = useStyles();
+  const [searchForm, setSearchForm] = useState(true);
 
-  return (
-    <Container component="main" maxWidth="md">
-      <form className={classes.form} noValidate>
-        <Typography component="h1" variant="h5">
-          Search our society's records
-        </Typography>
-        <NamePicker />
-        <LifeEventPicker />
-        <RelationshipPicker />
-        <CategoryPicker />
-        <div style={{ marginTop: 50 }}>
-          <Button variant="outlined" color="primary" type="submit" value="Submit">
-            Submit
-          </Button>
-        </div>
-      </form>
-    </Container>
-  );
+  if (searchForm) {
+    return (
+      <Container component="main" maxWidth="md">
+        <form className={classes.form} noValidate>
+          <Typography component="h1" variant="h5">
+            Search our society's records
+          </Typography>
+          <NamePicker />
+          <LifeEventPicker />
+          <RelationshipPicker />
+          <CategoryPicker />
+          <div style={{ marginTop: 50 }}>
+            <label onClick={() => setSearchForm(!searchForm)}>
+              <Button variant="outlined" color="primary" type="submit" value="Submit">
+                Search
+              </Button>
+            </label>
+          </div>
+        </form>
+      </Container>
+    );
+  }
+
+  return <SearchResults />;
```

Comment:
I think of the SearchForm and SearchResults and two fundamentally different pages. So it seems weird to have them both rendered by the same component.

Also, when this actually gets put together with server-side searching, the flow is going to be something like:

- SearchForm gets filled out by user
- SearchForm.submit sends request to server
- Server responds
- User is taken to separate page to view results.

It seems like this whole journey should be orchestrated by a higher-level component. So I'd probably do something like:

function Search() {
  return searchResults ?
    <SearchResults/>
    :
    <SearchForm/>
}

Which, at the end of the day, doesn't really change much about what you've done here. :-) It just moves a few lines of code to another file.

Reply:
Updated thanks! Sorry, it's still new to me, I guess I am still accustomed in rendering everything with a class in App.js.
But for now, I guess keep it as it is to demo the search button?

Reply:
It actually would be very easy to change, but you can keep it this way if you'd prefer. Here's the solution:
(For now, I'm just using a boolean to track whether the results page is visible. Ultimately we'd want to use an actual response from the server.)
```
export default function Search(): JSX.Element {
  const [results, setResults] = useState(false);
  return { results ?
    (<SearchResults onClose={() => setResults(false)} />) :
    (<SearchForm onSubmit={() => setResults(true)} />)
  };
}
```
It should be this way because you want a  component above the other two that will manage communication with the server and passing state between the two other components. The actual communication with the server will happen in this component, and the results will be passed along. Something like this:
```
useEffect(() => {
  const results = await axios.get(`/search?{query}`);
  setResults(results);
}
```
and then the results would be sent to the search results page as
`<SearchForm results={results} />`

As you can see, you're doing all the hard parts now with that detailed form work. Connecting it to the server will be pretty easy if we structure things right from the beginning. :-) I'll walk you through that when you're ready.

Reply:
Thanks for the explanation! I think I got it :) 
Since we already have a Search component, I will name it SearchPage if that is ok, and apply the above changes. Much appreciated!

Reply:
I added an any type to props, since booleans will get an error of 'This expression is not callable. Type 'Boolean' has no call signatures.ts(2349)'
And I used onClick Button for SearchForm, since onSubmit will not pass the props to render SearchResults. 
```
export default function SearchForm(props: { setResults: any }): JSX.Element {
  const { setResults } = props;
  const classes = useStyles();

  return (
    <Container component="main" maxWidth="md">
      <form className={classes.form} noValidate>
        <Typography component="h1" variant="h5">
          Search our society&apos;s records
        </Typography>
        <NamePicker />
        <LifeEventPicker />
        <RelationshipPicker />
        <CategoryPicker />
        <Box mt={5}>
          <Button
            variant="outlined"
            color="primary"
            type="submit"
            value="Submit"
            // onSubmit={() => setResults(true)}
            onClick={() => setResults(true)}
          >
            Search
          </Button>
        </Box>
      </form>
    </Container>
  );
}
```